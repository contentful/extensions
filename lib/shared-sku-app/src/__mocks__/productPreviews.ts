import { Product } from '../interfaces';

const productPreviews: Product[] = [
  {
    id: '28a7442d',
    image: 'https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-1.jpg',
    name: 'Dress Twin-Set rose',
    sku: 'M0E20130820E90Z',
    externalLink: 'https://mc.commercetools.com/example-project/products/28a7442d/general'
  },
  {
    id: '4e5446e8',
    image: 'https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-2.jpg',
    name: 'Michael Kors – Shopper “Jet Set Travel”',
    sku: 'A0E2300FX102203',
    externalLink: 'https://mc.commercetools.com/example-project/products/4e5446e8/general'
  },
  {
    id: 'db3c234l',
    image: 'https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-3.jpg',
    name: 'Shirt “Jenny“ Polo Ralph Lauren white',
    sku: 'M0E21300900DZN7',
    externalLink: 'https://mc.commercetools.com/example-project/products/db3c234l/general'
  }
];

export default productPreviews;
