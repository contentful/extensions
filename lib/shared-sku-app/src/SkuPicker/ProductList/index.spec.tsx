import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Props, ProductList } from '.';
import productPreviews from '../../__mocks__/productPreviews';

const defaultProps: Props = {
  products: productPreviews,
  selectProduct: jest.fn(),
  selectedSKUs: []
};

const renderComponent = (props: Props) => {
  return render(<ProductList {...props} />);
};

describe('ProductList', () => {
  afterEach(cleanup);

  it('should render successfully with no items selected', async () => {
    const component = renderComponent(defaultProps);
    expect(component.container).toMatchSnapshot();
  });

  it('should render successfully with selected items', () => {
    const component = renderComponent({
      ...defaultProps,
      selectedSKUs: [productPreviews[1].sku]
    });
    expect(component.container).toMatchSnapshot();
  });
});
