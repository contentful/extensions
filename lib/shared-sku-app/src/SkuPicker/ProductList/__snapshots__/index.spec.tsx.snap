// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render successfully with no items selected 1`] = `
<div>
  <div
    class="css-almiqb"
  >
    <div
      class="css-zkv16y"
    >
      <div
        aria-checked="false"
        class="css-1qso258 "
        data-test-id="product-preview-M0E20130820E90Z"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-id)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-id"
            >
              <rect
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation1"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-19cpqqr"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-1.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Dress Twin-Set rose
        </p>
        <p
          class="css-1kn9x3e"
        >
          M0E20130820E90Z
        </p>
      </div>
    </div>
    <div
      class="css-zkv16y"
    >
      <div
        aria-checked="false"
        class="css-1qso258 "
        data-test-id="product-preview-A0E2300FX102203"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-id)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-id"
            >
              <rect
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation1"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-19cpqqr"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-2.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Michael Kors – Shopper “Jet Set Travel”
        </p>
        <p
          class="css-1kn9x3e"
        >
          A0E2300FX102203
        </p>
      </div>
    </div>
    <div
      class="css-zkv16y"
    >
      <div
        aria-checked="false"
        class="css-1qso258 "
        data-test-id="product-preview-M0E21300900DZN7"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-id)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-id"
            >
              <rect
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation1"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-19cpqqr"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-3.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Shirt “Jenny“ Polo Ralph Lauren white
        </p>
        <p
          class="css-1kn9x3e"
        >
          M0E21300900DZN7
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductList should render successfully with selected items 1`] = `
<div>
  <div
    class="css-almiqb"
  >
    <div
      class="css-zkv16y"
    >
      <div
        aria-checked="false"
        class="css-1qso258 "
        data-test-id="product-preview-M0E20130820E90Z"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-id)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-id"
            >
              <rect
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation1"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-19cpqqr"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-1.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Dress Twin-Set rose
        </p>
        <p
          class="css-1kn9x3e"
        >
          M0E20130820E90Z
        </p>
      </div>
    </div>
    <div
      class="css-zkv16y"
    >
      <div
        aria-checked="true"
        class="css-1qso258 css-1rtg1va"
        data-test-id="product-preview-A0E2300FX102203"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-id)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-id"
            >
              <rect
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation1"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-19cpqqr"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-2.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1bpg4ef"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Michael Kors – Shopper “Jet Set Travel”
        </p>
        <p
          class="css-1kn9x3e"
        >
          A0E2300FX102203
        </p>
      </div>
    </div>
    <div
      class="css-zkv16y"
    >
      <div
        aria-checked="false"
        class="css-1qso258 "
        data-test-id="product-preview-M0E21300900DZN7"
        role="switch"
        tabindex="-1"
      >
        <svg
          aria-label="Loading component..."
          class="SkeletonContainer__SkeletonContainer___23jiu css-7uqnex"
          data-test-id="cf-ui-skeleton-form"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          width="100%"
        >
          <title>
            Loading component...
          </title>
          <rect
            clip-path="url(#cf-ui-skeleton-clip-id)"
            height="100%"
            style="fill: url(#cf-ui-skeleton-clip-gradient);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="cf-ui-skeleton-clip-id"
            >
              <rect
                height="290"
                rx="0"
                ry="0"
                width="400"
              />
            </clippath>
            <lineargradient
              id="cf-ui-skeleton-clip-gradient"
            >
              <stop
                offset="0%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  dur="2s"
                  id="animation1"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#f7f9fa"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.25s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#e5ebed"
                stop-opacity="1"
              >
                <animate
                  attributeName="stop-color"
                  begin="animation1.begin+0.5s"
                  dur="2s"
                  repeatCount="indefinite"
                  values="#e5ebed; #f7f9fa; #e5ebed"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
        <div
          class="css-19cpqqr"
        >
          <img
            alt="product preview"
            class="css-1ddetr"
            data-test-id="image"
            src="https://s3-eu-west-1.amazonaws.com/commercetools-example/products/picture-3.jpg"
            style="display: none;"
          />
          <img
            alt="check"
            class="css-1imdisq"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iNDQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGcgZmlsbD0iIzMwNzJCRSIgZmlsbC1ydWxlPSJldmVub2RkIj48Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjEiIGN4PSIyMiIgY3k9IjIyIiByPSIyMiIvPjxwYXRoIGQ9Ik0zMS42ODkgMTYuNjE0TDI5LjU1NSAxNC41IDE5LjAyNCAyNS4yNzVsLTQuNjItNC40MDYtMi4wOTIgMi4xNTMgNi43NSA2LjQ3OCAxMi42MjctMTIuODg2eiIgZmlsbC1ydWxlPSJub256ZXJvIi8+PC9nPjwvc3ZnPg=="
          />
        </div>
        <p
          class="css-182natc"
        >
          Shirt “Jenny“ Polo Ralph Lauren white
        </p>
        <p
          class="css-1kn9x3e"
        >
          M0E21300900DZN7
        </p>
      </div>
    </div>
  </div>
</div>
`;
